packages/experimental/AppearanceAdditions/ios/FRNAppearanceAdditions.m

```diff
@@ -103,11 +97,31 @@ + (BOOL)requiresMainQueueSetup {
     return @[ @"appearanceChanged" ];
 }
 
+- (dispatch_queue_t)methodQueue
+{
+    return dispatch_get_main_queue();
+}
+
 - (void)startObserving {
     _hasListeners = YES;
-    _horizontalSizeClass = RCTHorizontalSizeClassPreference(nil);
-    _userInterfaceLevel = RCTUserInterfaceLevelPreference(nil);
-    _accessibilityContrastOption = RCTAccessibilityContrastPreference(nil);
+    
+    // Note that [UITraitCollection currentTraitCollection] always returns the same default trait collection,
+    // presumably because FRNAppearanceAdditions isn't a view, so it never gets updated with the right traitCollection
+    // (which happens when a view gets added to the view hierachy). In order to get the right trait collection,
+    // we need to access a view that's been added to the view hierarchy
+    UIViewController *viewControllerWithInitialTraitCollection = RCTPresentedViewController();
```

Comment:
I have a vague feeling this method might have a diff in it between React Native and React Native macOS. Might be worth double checking

Reply:
These look the same to me but let me know if I'm not comparing the right things

https://github.com/facebook/react-native/blob/main/React/Base/RCTUtils.m
https://github.com/microsoft/react-native-macos/blob/main/React/Base/RCTUtils.m#L612

Reply:
I guess not. Thanks for confirming!